src/vmm/src/acpi/mod.rs

```diff
@@ -272,15 +272,15 @@ pub mod tests {
             resource_allocator: &mut vmm.resource_allocator,
         };
 
-        let mut sdt = MockSdt(vec![0; usize::try_from(SYSTEM_MEM_SIZE).unwrap()]);
+        let mut sdt = MockSdt(vec![0; usize::try_from(SYSTEM_MEM_SIZE - 4096).unwrap()]);
         let err = writer.write_acpi_table(&mut sdt).unwrap_err();
         assert!(
             matches!(
                 err,
                 AcpiError::AcpiTables(acpi_tables::AcpiError::GuestMemory(
                     vm_memory::GuestMemoryError::PartialBuffer {
-                        expected: 263168,  // SYSTEM_MEM_SIZE
-                        completed: 259072  // SYSTEM_MEM_SIZE - 4096
+                        expected: 259072,  // SYSTEM_MEM_SIZE - 4096
+                        completed: 254976  // SYSTEM_MEM_SIZE - 8192
```

Comment:
Maybe use `SYSTEM_MEM_SIZE` directly? + instead of 4096 can use `MMIO_LEN`.

Reply:
Looks like I can't do the subtraction needed during the instantiation of the struct. Would you prefer if I shifted it to a variable above the assert?

Reply:
Hmm, when I tried extracting it to a variable, it didn't compare the values inside, so I've left it as is for now.